### PR TITLE
test(actions): rebase action

### PR DESCRIPTION
**Pull request by CamClrt**

### CODE REVIEW

**Comment by jd** [comment]
```
INFO     Context:actions_runner.py:443 action evaluation: `rebase` executed: run/neutral -> success
```
If you read the code, the only case where rebase returns success are either:
1. the PR has been rebased
2. the PR is up to date

Clearly 1. did not happen, so it must be 2.

The code does:

```
if await ctxt.is_behind:
```

I'm not sure why it'd think the PR is behind. Hint @sileht?

File: mergify_engine/tests/functional/actions/test_rebase.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        p, commits = await self.create_pr()
+        pr_initial_sha = commits[-1].sha
+
+        await self.git("checkout", ahead_branch)
+
+        open(self.git.tmp + "/an_other_file", "wb").close()
+        await self.git("add", "an_other_file")
+
+        await self.git("commit", "--no-edit", "-m", "commit ahead")
+        await self.git("push", "--quiet", "main", self.master_branch_name)
```
Comment (sileht) [inline mergify_engine/tests/functional/actions/test_rebase.py:53]: The `wait_for("push")` should be here.

Comment (CamClrt) [reply mergify_engine/tests/functional/actions/test_rebase.py:53]: That's my problem when I put the `wait_for()` line 49 or line 51, I raise this exception:

```
raise Exception(
            f"Never got event `{event_type}` with payload `{expected_payload}` (timeout)"
        )
E       Exception: Never got event `push` with payload `{'ref': 'refs/heads/20210120134511/test_rebase_ok/main/head'}` (timeout)

mergify_engine/tests/functional/base.py:183: Exception
```

Comment (jd) [reply mergify_engine/tests/functional/actions/test_rebase.py:53]: Why are you trying L49 or L51 while @sileht suggested L53? 🤔 
Did you try L53?

Comment (CamClrt) [reply mergify_engine/tests/functional/actions/test_rebase.py:53]: same

Comment (CamClrt) [reply mergify_engine/tests/functional/actions/test_rebase.py:52]: I have just refreshed my code if you want more details 👀

File: mergify_engine/tests/functional/actions/test_rebase.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            "pull_request_rules": [
+                {
+                    "name": "rebase",
+                    "conditions": [f"base={self.master_branch_name}"],
+                    "actions": {"rebase": {}},
+                }
+            ]
+        }
+
+        ahead_branch = self.get_full_branch_name("main/head")
```
Comment (GuillaumeOj) [inline mergify_engine/tests/functional/actions/test_rebase.py:33]: My bad, this is useless

File: mergify_engine/tests/functional/actions/test_rebase.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            yaml.dump(rules),
+            test_branches=[
+                ahead_branch,
+            ],
+        )
+
+        p, commits = await self.create_pr()
+        pr_initial_sha = commits[-1].sha
+
+        await self.git("checkout", ahead_branch)
```
Comment (GuillaumeOj) [inline mergify_engine/tests/functional/actions/test_rebase.py:45]:
```suggestion
        await self.git("checkout", self.master_branch_name)
```
Is what you want

File: mergify_engine/tests/functional/actions/test_rebase.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+        open(self.git.tmp + "/an_other_file", "wb").close()
+        await self.git("add", "an_other_file")
+
+        await self.git("commit", "--no-edit", "-m", "commit ahead")
+        await self.wait_for("push", {"ref": f"refs/heads/{ahead_branch}"})
+        await self.git("push", "--quiet", "main", self.master_branch_name)
```
Comment (GuillaumeOj) [inline mergify_engine/tests/functional/actions/test_rebase.py:52]:
```suggestion
        await self.git("push", "--quiet", "main", self.master_branch_name)
        await self.wait_for("push", {"ref": f"refs/heads/self.master_branch_name"})
```
Is better

**Review by GuillaumeOj — CHANGES_REQUESTED** [review]

**Comment by CamClrt** [comment]
:) !

**Review by jd — APPROVED** [review]

**Pull request merged**